api/v1alpha1/externaldns_types.go

```diff
@@ -510,7 +515,11 @@ type ExternalDNSServiceSourceOptions struct {
 }
 
 type ExternalDNSOpenShiftRouteOptions struct {
-	// If source is openshift-route then you can pass the ingress controller name. Based on this name external-dns will select the respective router.
+	// RouterName is the name of a router (AKA ingress controller) as
+	// reported in Route.status.ingress[].routerName.  External-dns will use
+	// the canonical hostname of the router identified by this name when
+	// publishing records for a given route.
```

Comment:
@deads2k, I tried to clarify this per your review.

Reply:
> @deads2k, I tried to clarify this per your review.

looks good.

Reply:
actually, are they guaranteed to all be the same?

Reply:
I don't understand your question.  When external-dns publishes a record for a route, it publishes a CNAME record with the router's canonical hostname as the target.  This field in the externaldns CRD specifies the name of the router.  The canonical hostname should be the same for all routes exposed by a given external-dns instance.  (If two routes are served by the same router but have different canonical hostnames in their status, then something weird has happened.)